apps/web/pages/api/verify-booking-token.ts

```diff
@@ -0,0 +1,91 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { authenticator } from "otplib";
+import { z } from "zod";
+
+import { defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+import { UserPermissionRole } from "@calcom/prisma/enums";
+import { TRPCError } from "@calcom/trpc/server";
+import { createContext } from "@calcom/trpc/server/createContext";
+import { bookingsRouter } from "@calcom/trpc/server/routers/viewer/bookings/_router";
+import type { UserProfile } from "@calcom/types/UserProfile";
+
+enum DirectAction {
+  ACCEPT = "accept",
+  REJECT = "reject",
+}
+
+const querySchema = z.object({
+  action: z.nativeEnum(DirectAction),
+  token: z.string(),
+  reason: z.string().optional(),
+  bookingUid: z.string(),
+  userId: z.string(),
```

Comment:
As tokens are unique then we can use them to identify the booking. We don't need the UID.

```suggestion
  action: z.nativeEnum(DirectAction),
  token: z.string(),
  reason: z.string().optional(),
  userId: z.string(),
```

Reply:
I think we need it. If the token is not valid, we don't get the booking UID, but the UID is needed to redirect to the /uid/404 page.